tests/relay/server.js

```diff
@@ -0,0 +1,185 @@
+
+'use strict';
+
+const ngrok = require('@ngrok/ngrok');
+const axios = require('axios').default;
+const express = require('express');
+const cors = require('cors');
+
+/** @type {Map<string, any[]>} */
+const receivedMessages = new Map();
+
+/**
+ * @returns {{
+ *    apiKeySid: string;
+ *    apiKeySecret: string;
+ *    accountSid: string;
+ * }}
+ */
+const getEnvVars = () => {
+  const apiKeySid = process.env['API_KEY_SID'];
+  if (typeof apiKeySid !== 'string') {
+    throw new Error('API_KEY_SID env var not defined.');
+  }
+
+  const apiKeySecret = process.env['API_KEY_SECRET'];
+  if (typeof apiKeySecret !== 'string') {
+    throw new Error('API_KEY_SECRET env var not defined.');
+  }
+
+  const accountSid = process.env['ACCOUNT_SID'];
+  if (typeof accountSid !== 'string') {
+    throw new Error('ACCOUNT_SID env var not defined.');
+  }
+
+  return { apiKeySid, apiKeySecret, accountSid };
+};
+
+/**
+ * @typedef {(callSid: string, callbackUrl: string) => Promise<import('axios').AxiosResponse>} CreateSubscription
+ */
+
+/**
+ * @typedef {(callSid: string) => Promise<import('axios').AxiosResponse>} SendMessage
+ */
+
+/**
+ * @param {string} accountSid
+ * @param {string} apiKeySid
+ * @param {string} apiKeySecret
+ */
+const createTwilioUserDefinedMessageActions = (accountSid, apiKeySid, apiKeySecret) => {
+  const baseUrl = 'https://api.twilio.com/2010-04-01';
+  const extendedUrl = `${baseUrl}/Accounts/${accountSid}/Calls`;
+
+  /**
+   * @type {CreateSubscription}
+   */
+  const createSubscription = (callSid, callbackUrl) => {
+    const url =
+      `${extendedUrl}/${callSid}/UserDefinedMessageSubscriptions.json`;
+
+    const params = new URLSearchParams();
+    params.append('Method', 'POST');
+    params.append('Callback', callbackUrl);
+
+    return axios.post(url, params, {
+      auth: {
+        username: apiKeySid,
+        password: apiKeySecret,
+      },
+    });
+  };
+
+  /**
+   * @type {SendMessage}
+   */
+  const sendMessage = (callSid) => {
+    const url =
+      `${extendedUrl}/${callSid}/UserDefinedMessages.json`;
+
+    const params = new URLSearchParams();
+    params.append('Content', JSON.stringify({ message: 'ahoy, world!' }));
+
+    return axios.post(url, params, {
+      auth: {
+        username: apiKeySid,
+        password: apiKeySecret,
+      },
+    });
+  };
+
+  return { createSubscription, sendMessage };
+}
+
+/**
+ * @param {CreateSubscription} createSubscription
+ * @param {SendMessage} sendMessage
+ */
+const createExpressApp = (createSubscription, sendMessage) => {
+  const app = express();
+
+  app.use(cors());
+  app.use(express.json());
+  app.use(express.urlencoded({ extended: true }));
+
+  app.post('/create-subscription', async (req, res) => {
+    const { CallSid } = req.body;
+    if (typeof CallSid !== 'string') {
+      res.sendStatus(400);
+      throw new Error('CallSid is not of type string');
+    }
+    const callbackUrl = `${app.listener.url()}/receive-message`;
```

Comment:
Is `app.listener.url()` always unique? I'm assuming we can run tests in parallel right?

Reply:
Should always be unique, it's the `ngrok` URL. So as long as `ngrok` makes a unique URL each time, we're good.